### PR TITLE
Add support for Rails 7.1

### PR DESCRIPTION
**Pull request by rkrage**
@igor-alexandrov (lol sorry I just saw that you opened a PR as well) 
@kevinrobayna

Note that this also adds a requirement for Ruby 3 and removes EOL Postgres versions from the build matrix

### CODE REVIEW

File: lib/pg_party/hacks/postgresql_database_tasks.rb [hunk 1]
```diff
@@ -8,18 +8,14 @@ def run_cmd(cmd, args, action)
...
-          []
-        end
+        partitions = ActiveRecord::Base.connection.select_values(<<-SQL)
+          SELECT
+            inhrelid::regclass::text
+          FROM
+            pg_inherits
+          JOIN pg_class AS p ON inhparent = p.oid
+          WHERE p.relkind = 'p'
+        SQL
```
Comment (rkrage) [inline lib/pg_party/hacks/postgresql_database_tasks.rb:18]: @igor-alexandrov I fixed the failing test by removing it 😆 

I don't think it ever really made sense to rescue an error here anyways

Comment (igor-alexandrov) [reply lib/pg_party/hacks/postgresql_database_tasks.rb:18]: Yeaahhh! I love this approach, dude!

**Comment by igor-alexandrov** [comment]
@rkrage this looks good to me!

**Comment by igor-alexandrov** [comment]
Maybe it would be good to add **Requirements** section to the Readme and announce that PgParty works with Rails 6.1 and above and PostgreSQL > 10.

**Comment by rkrage** [comment]
> Maybe it would be good to add Requirements section to the Readme and announce that PgParty works with Rails 6.1 and above and PostgreSQL > 10.

Yup that's probably a good idea. I have a few other things I'd like to add to the readme as well, so I'll follow up with another commit after I merge this

**Pull request merged**